### PR TITLE
Add rubocop rake task

### PR DESCRIPTION
**Pull request by bunnymatic**
Problem
-------

More dog-fooding.  We want to promote the use of our carbonfive/c5-conventions
rubocop file.  Let's use it here.

Solution
--------

* Copy the carbonfive/c5-conventions `.rubocop.yml` into place
* Remove `rubocop-performance` because we don't *really* need it and
it'd require adding another gem dev depenency - seems overkill for it's
possible benefit on *this* project
* Add `lib/tasks` to `rakelib`
* Add `rubocop.rake` file to get the `rubocop` and `rubocop:autocorrect`
tasks
* Run rubocop:autocorrect which cleaned a few things in bin/raygun

NOTE
----

This repo is *still* not rubocop clean.  There are several
not-autofixable issues that could (should?) be addressed once this is
approved.

### CODE REVIEW

**Comment by bunnymatic** [comment]
Fixes #152

**Comment by bunnymatic** [comment]
Note: this PR is currently a branch on a branch (#147) just to keep the diff clean.  PR #147 would have to come first, and it is currently waiting on #151.  When these are all approved we can merge in order.

File: .rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,104 @@
+# Configuration hierarchy:
+#
+# 1. Rubocop defaults
+# 2. Carbon Five defaults (this file)
+# 3. Project overrides
+#
+# See http://rubocop.readthedocs.io/en/latest/configuration/#inheriting-configuration-from-a-remote-url for details.
+#
```
Comment (mattbrictson) [inline .rubocop.yml]: I think we should remove this since it only relevant to the `inherit_from:` directive, which we are not using here.

File: lib/tasks/rubocop.rake [hunk 2]
```diff
@@ -0,0 +1,3 @@
+require "rubocop/rake_task"
+
+RuboCop::RakeTask.new
```
Comment (mattbrictson) [inline lib/tasks/rubocop.rake]: I'd prefer we inline this in the `Rakefile` because putting this in `lib` implies that it is meant to be packaged and distributed as part of the raygun gem.

File: raygun.gemspec [hunk 3]
```diff
@@ -23,4 +23,5 @@ Gem::Specification.new do |gem|
 
   gem.add_development_dependency "bundler", "~> 2.0"
   gem.add_development_dependency "rake", "~> 13.0"
+  gem.add_development_dependency "rubocop", "~> 0.79"
```
Comment (mattbrictson) [inline raygun.gemspec]: Since we are not committing a `Gemfile.lock` I think we should pin rubocop here. New versions of rubocop often contain breaking changes, unfortunately.

```suggestion
  gem.add_development_dependency "rubocop", "0.79.0"
```

File: bin/raygun [hunk 4]
```diff
@@ -1,10 +1,10 @@
 #!/usr/bin/env ruby
 
-File.expand_path('../../lib', __FILE__).tap do |lib|
+File.expand_path("../lib", __dir__).tap do |lib|
```
Comment (mattbrictson) [inline bin/raygun]: 👍

File: .rubocop.yml [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+AllCops:
+  DisplayCopNames: true
+  DisplayStyleGuide: true
```
Comment (mattbrictson) [inline .rubocop.yml]: Non-blocking: It would be good to explicitly specify the version of Ruby compatibility that we are aiming for here:

```suggestion
  DisplayStyleGuide: true
  TargetRubyVersion: 2.4
```

We can do this in a separate PR when we fix the other non-autocorrectable RuboCop issues.

Comment (bunnymatic) [reply .rubocop.yml]: Interestingly, rubocop complains that the gemspec required version doesn't match that of the rubocop.yml so fixing this will fix a rubocop warning.

File: .rubocop.yml [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+Style/ModuleFunction:
+  EnforcedStyle: extend_self
+
+Style/MutableConstant:
+  Enabled: false
+
+Style/PreferredHashMethods:
+  Enabled: false
+
+Style/SpecialGlobalVars:
```
Comment (bunnymatic) [inline .rubocop.yml]: Note: Disabled this cop.  It is not in the `c5-conventions` version yet (expect a PR soon). 

It prefers (and auto corrects) the use of globals like `$?` with `$CHILD_STATUS` etc.  
The problem is *if* you autocorrect, and try to run the code, unless you've already `require 'english'` in your code, this will cause the code to fail.

Before we enforce this, we should think about how to set it and also if we want to `require 'english'` just so we can get rubocop to do it's business.  For now, I decided to leave it out.

More reading: https://github.com/rubocop-hq/rubocop/issues/1747

For the future, we could configure it to prefer or allow the `perl_names` which are the non-english, shortcut versions that we are already using like `$?` and `$&` etc.

**Comment by bunnymatic** [comment]
Followup PR https://github.com/carbonfive/raygun/pull/153 addresses remaining non-auto-correct able issues.
